util/perl/OpenSSL/paramnames.pm

```diff
@@ -691,29 +715,36 @@ sub generate_decoder_from_trie {
         print ")) {\n";
         trie_matched($field, $num, $indent1, $indent2);
         printf "%s}\n", $indent0;
+        output_endifdef($ifdefs->{$field});
         return;
     }
 
     printf "%sswitch(s\[%d\]) {\n", $indent0, $n;
     printf "%sdefault:\n", $indent0;
     for my $l (sort keys %$trieref) {
+        $need_break = 0;
         if ($l eq 'val') {
             $field = $identmap->{$trieref->{'val'}};
             my $num = $concat_num->{$field};
             printf "%sbreak;\n", $indent1;
+            output_ifdef($ifdefs->{$field});
             printf "%scase '\\0':\n", $indent0;
             trie_matched($field, $num, $indent1, $indent2);
-        } else {
+            output_endifdef($ifdefs->{$field});
+       } else {
             printf "%sbreak;\n", $indent1;
             printf "%scase '%s':", $indent0, $l;
             if (not $case_sensitive) {
                 print "   case '-':" if ($l eq '_');
                 printf "   case '%s':", uc $l if ($l =~ /[a-z]/);
             }
             print "\n";
-            generate_decoder_from_trie($n + 1, $trieref->{$l}, $identmap, $concat_num);
+            generate_decoder_from_trie($n + 1, $trieref->{$l}, $identmap, $concat_num, $ifdefs);
         }
     }
+    if ($need_break) {
+        printf "%sbreak;\n", $indent1;
```

Comment:
Can you explain this change. Not sure I quite understand this.

Reply:
The generator was producing switch statements that ended with a case block without a statement which produced compiler warnings.  I.e. code was produced like this:

```c
switch(…) {
…
case 'x':
#ifdef XYZZY
    r->x = (OSSL_PARAM *)p;
#endif
}
```

This changes puts an otherwise redundant `break` at the end:
```c
switch(…) {
…
case 'x':
#ifdef XYZZY
    r->x = (OSSL_PARAM *)p;
#endif
    break;
}
```

Normally the generator omits anything that isn't strictly required and that final break is not required.  But now it sometimes is.